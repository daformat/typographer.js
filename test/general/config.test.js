import typographer from '../../dist/typographer.umd';

test('Calling typographer without config works', () => {
  expect(
    typographer(
      'Bonjour typographer!'
    )
  ).toBe('Bonjour typographer!')
});

test('Calling typographer with a string returns a string', () => {
  expect(
    typeof typographer(
      'Bonjour typographer!'
    )
  ).toBe('string')
})

test('Calling typographer with a HTMLElement modifies the element in DOM', () => {
  document.body.innerHTML =
    '<div id="typographer">' +
    '  <p>Bonjour typographer !</p>' +
    '</div>';

  const element = document.querySelector('#typographer');

  typographer(
    element,
    {
      locale: 'en'
    }
  );

  expect(
    document.body
  ).toMatchSnapshot()
})
