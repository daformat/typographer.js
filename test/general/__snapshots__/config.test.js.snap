// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calling typographer with a HTMLElement modifies the element in DOM 1`] = `
<body>
  <div
    id="typographer"
  >
      
    <p>
      Bonjour typographer!
    </p>
  </div>
</body>
`;
